sep53/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+use defuse_crypto::{CryptoHash, Curve, Ed25519, Payload, SignedPayload, serde::AsCurve};
+use impl_tools::autoimpl;
+use near_sdk::{env, near};
+use serde_with::serde_as;
+
+/// See [SEP-53](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0053.md)
+#[near(serializers = [json])]
+#[serde(rename_all = "camelCase")]
+#[derive(Debug, Clone)]
+#[must_use]
+pub struct Sep53Payload {
+    pub message: Vec<u8>,
+}
+
+impl Sep53Payload {
+    #[inline]
+    pub const fn new(message: Vec<u8>) -> Self {
+        Self { message }
+    }
+
+    #[inline]
+    pub fn prehash(&self) -> Vec<u8> {
+        b"Stellar Signed Message:\n"
+            .iter()
+            .copied()
+            .chain(self.message.as_slice().iter().copied())
+            .collect()
```

Comment:
This would be more efficient:
```suggestion
        [b"Stellar Signed Message:\n".as_slice(), self.message.as_slice()].concat()
```